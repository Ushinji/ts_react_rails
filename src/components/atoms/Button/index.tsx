import * as React from 'react';
import styled from 'styled-components';

const StyledButton = styled.button`
  display: inline-block;
  font-size: 18px;
  border-radius: 8px;
  text-align: center;
  text-decoration: none;
  background-color: #337ab7;
  color: #ffffff;

  &:hover {
    text-decoration: none;
    color: #ffffff;
    background-color: #286090;
    cursor: pointer;
  }
`;

const Spinner = styled.div`
  border: 2px solid #dbdbdb;
  border-radius: 290486px;
  border-right-color: transparent;
  border-top-color: transparent;
  display: block;
  position: relative;
  height: 1em;
  width: 1em;

  animation: spinAround 1s infinite linear;
  @-webkit-keyframes spinAround {
    from {
      -webkit-transform: rotate(0deg);
      transform: rotate(0deg);
    }
    to {
      -webkit-transform: rotate(359deg);
      transform: rotate(359deg);
    }
  }
  @keyframes spinAround {
    from {
      -webkit-transform: rotate(0deg);
      transform: rotate(0deg);
    }
    to {
      -webkit-transform: rotate(359deg);
      transform: rotate(359deg);
    }
  }
`;

const Button: React.FC<
  React.ButtonHTMLAttributes<HTMLButtonElement>
> = props => {
  return (
    <StyledButton {...props}>
      {props.children}
      <Spinner />
    </StyledButton>
  );
};

export default Button;
