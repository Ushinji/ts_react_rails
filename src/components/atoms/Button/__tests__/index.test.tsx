import * as React from 'react';
import { render } from 'react-testing-library';
import Button from '..';

describe('atoms/Button', () => {
  it('Snapshot test', () => {
    const handleClick = () => {};
    const { asFragment } = render(
      <Button onClick={handleClick}>TEST_BUTTON</Button>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
