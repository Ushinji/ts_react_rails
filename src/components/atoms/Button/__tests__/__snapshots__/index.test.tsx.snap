// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Button Snapshot test 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 18px;
  border-radius: 8px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #3173dc;
  color: #ffffff;
  padding: 8px 16px;
  width: 100%;
  -webkit-transition: 0.2s;
  transition: 0.2s;
  outline: none;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #ffffff;
  background-color: #266cda;
  cursor: pointer;
}

.c0:hover:disabled {
  cursor: not-allowed;
}

.c0:disabled,
.c0:disabled:hover {
  opacity: 0.5;
  cursor: not-allowed;
}

<button
    class="c0"
  >
    TEST_BUTTON
  </button>
</DocumentFragment>
`;
